packages/ui-react/src/containers/ModalProvider/useModal.ts

```diff
@@ -0,0 +1,45 @@
+import { type RefObject, useContext, useEffect, useRef } from 'react';
+import useEventCallback from '@jwp/ott-hooks-react/src/useEventCallback';
+
+import { ModalContext } from './ModalProvider';
+
+type Params = {
+  open: boolean;
+  onClose?: () => void;
+  modalRef: RefObject<HTMLElement>;
+  containerRef?: RefObject<HTMLElement>;
+};
+
+export const useModal = ({ open, onClose, modalRef, containerRef }: Params) => {
+  const modalId = useRef(String(Math.round(Math.random() * 1_000_000))).current;
+  const { openModal, closeModal, modals } = useContext(ModalContext);
+  const isOpen = modals.some((modal) => modal.modalId === modalId);
```

Comment:
If you make this into a callback, then you don't need to disable eslint (which is always a risk for when people add variables).
```javascript
const isOpen = useCallback(() => modals.some((modal) => modal.modalId === modalId), []);
```
```javascript
  useEffect(() => {
    if (open) {
      openModal(modalId, modalRef, onCloseCallback, containerRef);
    } else if (isOpen()) {
      closeModal(modalId, false);
    }
  }, [open]);
```

Reply:
Are you sure? 😉

Reply:
Sorry, `isOpen` should still be added as a dep indeed, which would still retrigger the effect. ☕ 😄